aggregators/1.0/modules/ROOT/pages/aggregators-group-example.adoc

```diff
@@ -0,0 +1,160 @@
+
+.Group-Based Aggregator Flow Example
+image::aggregator-group-example.png[Group-Based Aggregator Flow Example in Anypoint Studio Canvas]
+
+To test this example in your own environment, you must create the Mule application and test it with curl commands.
+
+
+== Create the Mule Application
+
+To create the Mule flow in Anypoint Studio:
+
+. In *Mule Palette*, select *HTTP Listener* and drag it onto the canvas. +
+The source initiates the flow by listening for incoming HTTP message attributes.
+. In *Connector configuration*, select *HTTP_Listener_config*.
+. Set *Path* to `/test`.
+. Drag a *Transform message* component to the right of *Listener*.
+. In the DataSense preview window of the *Transform message* component, add the following DataWeave expression, which transforms each HTTP request by assigning a `Source Name` as `"size-based"` and a `Source ID` as `random() as String`. The expression also obtains the `GroupId` value of the request based on `message.attributes['queryParams'].groupId`:
++
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/json
+---
+{
+    "Source Name": "group-based",
+    "GroupId": message.attributes['queryParams'].groupId,
+    "Source ID": (random() as String)
+}
+----
+[start=6]
+. Drag the *Group based aggregator* scope to the right of *Transform message*.
+. Set *Name* to `groupBasedAggregator`.
+. Set *Content* to `payload`, which is the expression that defines what to aggregate: in this case, the transformed HTTP requests.
+. Set *Group id* to `payload.GroupId`, which is the expression to evaluate for every new message received that gets the group ID where the element is aggregated.
+. Set *Group size* to `3`, which is the maximum size of elements to aggregate to the group with the group ID resolved.
+. Drag a *Logger* component into the *Incremental aggregation* route.
+. In the *Logger* configuration screen, set *Message* to `Doing incremental aggregation step.` +
+. Drag a *Logger* component into the *Aggregation complete* route.
+. Set *Message* to `Aggregation complete`.
+. Drag another *Logger* component to the right of the first *Logger* component in *Aggregation complete*.
+. Set *Message* to the following expression, which returns all the aggregated elements:
++
+[source,xml,linenums]
+----
+output application/json
+---
+payload
+----
+. Save the project.
+. In *Package Explorer*, click *Run > Run As > Mule Application*.
+. Test the app by sending the curl command `http://localhost:8086/test?groupId=1` twice. +
```

Comment:
I'm not sure if that's implied, but the curl is a POST (as configured in the XML below), so it will be like this
`curl -X POST http://localhost:8086/test\?groupId=1`